flexmeasures_openweathermap/cli/tests/test_get_forecasts.py

```diff
@@ -0,0 +1,53 @@
+from datetime import datetime, timedelta
+
+from flexmeasures.data.models.time_series import TimedBelief
+from flexmeasures.utils.time_utils import as_server_time, get_timezone
+
+from ..commands import collect_weather_data
+from ...utils import owm
+
+
+"""
+Useful resource: https://flask.palletsprojects.com/en/2.0.x/testing/#testing-cli-commands
+"""
+
+sensor_params = {"name": "wind_speed", "latitude": 30, "longitude": 40}
+
+
+def test_get_weather_forecasts_to_db(
+    app, fresh_db, monkeypatch, add_weather_sensors_fresh_db
+):
+    """ Test if we can process forecast and save them to the database."""
+    fresh_db.session.flush()
+    wind_sensor_id = add_weather_sensors_fresh_db["wind"].id
+
+    def mock_owm_response(api_key, location):
+        mock_date = datetime.now()
+        mock_date_tz_aware = as_server_time(
+            datetime.fromtimestamp(mock_date.timestamp(), tz=get_timezone())
+        ).replace(second=0, microsecond=0)
+        return mock_date_tz_aware, [
+            {"dt": mock_date.timestamp(), "temp": 40, "wind_speed": 100},
+            {
+                "dt": (mock_date + timedelta(hours=1)).timestamp(),
+                "temp": 42,
+                "wind_speed": 90,
+            },
+        ]
+
+    monkeypatch.setenv("OPENWEATHERMAP_API_KEY", "dummy")
+    monkeypatch.setattr(owm, "call_openweatherapi", mock_owm_response)
+
+    runner = app.test_cli_runner()
+    result = runner.invoke(collect_weather_data, ["--location", "33.484,126"])
```

Comment:
Slightly more robust would be to get the location from `add_weather_sensors_fresh_db["wind"]`. I noticed that the sensor was actually set up with four more latitude decimals (`latitude=33.4843866`), so the location here is slightly different. Why are the forecasts still attributed to the wind sensor with `wind_sensor_id`? An answer to that would be  good to add to the test docstring.

Reply:
The weather station was still close enough. I added a test where it is not close enough, and we fail to attribute the forecasts to a sensor.